itest/lnd_open_channel_test.go

```diff
@@ -621,3 +621,34 @@ func verifyCloseUpdate(chanUpdate *lnrpc.ChannelEventUpdate,
 
 	return nil
 }
+
+// testFundingExpiryBlocksOnPending checks that after an OpenChannel, and
+// before the funding transaction is confirmed, that the FundingExpiryBlocks
+// field of a PendingChannels decreases.
+func testFundingExpiryBlocksOnPending(ht *lntest.HarnessTest) {
+	alice, bob := ht.Alice, ht.Bob
+	param := lntest.OpenChannelParams{Amt: 100000}
+	update := ht.OpenChannelAssertPending(alice, bob, param)
+
+	// At this point, the channel's funding transaction will have been
+	// broadcast, but not confirmed. Alice and Bob's nodes should reflect
+	// this when queried via RPC. FundingExpiryBlock should decrease
+	// as blocks are mined, until the channel is confirmed. Empty blocks
+	// won't confirm the funding transaction, so let's mine a few empty
+	// blocks and verify the value of FundingExpiryBlock at each step.
+	const numEmptyBlocks = 3
+	for i := int32(0); i < numEmptyBlocks; i++ {
+		expectedVal := funding.MaxWaitNumBlocksFundingConf - i
+		pending := ht.AssertNumPendingOpenChannels(alice, 1)
+		require.Equal(ht, expectedVal, pending[0].FundingExpiryBlocks)
+		pending = ht.AssertNumPendingOpenChannels(bob, 1)
+		require.Equal(ht, expectedVal, pending[0].FundingExpiryBlocks)
+		ht.MineEmptyBlocks(1)
+	}
+
+	// Mine 1 block to confirm the funding transaction, and then close the
+	// channel.
+	ht.MineBlocksAndAssertNumTxes(1, 1)
+	chanPoint := lntest.ChanPointFromPendingUpdate(update)
+	ht.CloseChannel(alice, chanPoint)
```

Comment:
@yyforyongyu, I went for a slightly different approach. Still need to mine a block to gracefully close the channel, but I don't think we need to check that it transitions to open. This works on an isolated local run, and I'm testing if the complete `make itest` passes locally. Let me know if you have any thoughts. Thanks.

Reply:
You can always refer to the build results below to see the errors. And you can test it against different backend nodes as [documented here](https://github.com/lightningnetwork/lnd/blob/master/itest/README.md#additional-arguments). Our build runs against `btcd`, `neutrino` and multiple modes of `bitcoind` mixed with different database options.

As for why the previous test was flaky, it was because the method `ht.MineBlocks(1)` is used instead of `ht.MineBlocksAndAssertNumTxes(1)`. Because we use regtest, mining blocks can be very fast, thus the funding tx may have not entered the mempool yet but the block is already mined, which explains why the test failed for some of the backends.

The current approach works because `ht.MineBlocksAndAssertNumTxes` is used.